poet/sawtooth_poet/validator_registry/processor/handler.py

```diff
@@ -0,0 +1,333 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+import logging
+import hashlib
+import base64
+import json
+
+from sawtooth_signing import pbct as signing
+
+from sawtooth_sdk.processor.state import StateEntry
+from sawtooth_sdk.client.future import FutureTimeoutError
+from sawtooth_sdk.processor.exceptions import InvalidTransaction
+from sawtooth_sdk.processor.exceptions import InternalError
+from sawtooth_sdk.protobuf.transaction_pb2 import TransactionHeader
+
+from sawtooth_poet.protobuf.validator_registry_pb2 import \
+    ValidatorInfo
+from sawtooth_poet.protobuf.validator_registry_pb2 import \
+    ValidatorMap
+from sawtooth_poet.protobuf.validator_registry_pb2 import \
+    ValidatorRegistryPayload
+
+LOGGER = logging.getLogger(__name__)
+
+STATE_TIMEOUT_SEC = 10
+VAL_REG_NAMESPACE = \
+    hashlib.sha256("validator_registry".encode()).hexdigest()[0:6]
+
+
+def _get_address(key):
+    address = VAL_REG_NAMESPACE + hashlib.sha256(key.encode()).hexdigest()
+    return address
+
+
+def _get_validator_state(state, validator_id=None):
+    if validator_id is None:
+        address = _get_address('validator_map')
+        validator_state = ValidatorMap()
+    else:
+        validator_state = ValidatorInfo()
+        address = _get_address(validator_id)
+    try:
+        entries_list = state.get([address], timeout=STATE_TIMEOUT_SEC)
+    except FutureTimeoutError:
+        LOGGER.warning('Timeout occured on state.get([%s])', address)
+        raise InternalError('Unable to get {}'.format(address))
+
+    if len(entries_list) != 0:
+        validator_state.ParseFromString(entries_list[0].data)
+
+    return validator_state
+
+
+def _update_validator_state(state, validator_id, anti_sybil_id):
+    validator_map = _get_validator_state(state)
+    for entry in validator_map.entries:
+        if anti_sybil_id == entry.key:
+            # The validator's old signup info is stale and needs to be revoked
+            validator_info = _get_validator_state(state, validator_id)
+
+            validator_info.registered = "revoked"
+            validator_info = validator_info.SerializeToString()
+            _set_validator_info(state, validator_id, validator_info)
+        return
+
+    # This is a new validator that needs to be added to the map
+    validator_map.entries.add(key=anti_sybil_id, value=validator_id)
+    address = _get_address('validator_map')
```

Comment:
Can the code below that sets the map entry be refactored with the _set_validator_info method so that it is not repeated?